gnosis/cowswap
# Claim hooks pt 1 - useClaimCallback

# Summary

Updated `ClaimType` from a TS type to an enum, to be inline with contract
Added modified `useClaimCallback`

Example of a successful tx claiming `1080` https://rinkeby.etherscan.io/tx/0x90ad7c1ffc412c75286f91ae821c23397bbe0e2e75fdcf6e5961101c7b1f220e

![Screenshot from 2021-12-24 07-16-03](https://user-images.githubusercontent.com/43217/147349336-82cfb40b-1468-46c4-9ba3-28c3711a0c92.png)


  # To Test

0. Edit [this file](https://github.com/gnosis/cowswap/pull/2041/files#diff-3ae56d49d1cbb7b1aea829a94599d8ae0f20f9c5ec804ad13ad50b9a92ac31c8R72) with a different claim since I already claimed `1080` (no way to do that in the UI yet)
1. Connect wallet to rinkeby
2. Click on `claim vCOW`
3. Click on `claim` in the modal
* Your transaction should be sent

  # Background

*Optional: Give background information for changes you've made, that might be difficult to explain via comments*
